core/pta/device.c

```diff
@@ -88,14 +88,21 @@ static TEE_Result invoke_command(void *pSessionContext __unused,
 {
 	TEE_Result res = TEE_SUCCESS;
 	uint32_t rflags = 0;
+	/*
+	 * FIXME: this should also be true if CFG_RPMB_ANNOUNCE_PROBE_CAP is
+	 * enabled but the kernel does not support OP-TEE RPMB operations
+	 * (probe failed)
```

Comment:
Should we assume `PTA_CMD_GET_DEVICES` is always called before `PTA_CMD_GET_DEVICES_SUPP` and that `case PTA_CMD_GET_DEVICES_RPMB` may be called in between (which means kernel supports it new RPMB subsystem): using so, we could force `rflags |= TA_FLAG_DEVICE_ENUM_TEE_STORAGE_PRIVATE` when either `PTA_CMD_GET_DEVICES_RPMB` was not called or it failed on `tee_rpmb_reinit()`?

Reply:
I'm not sure we can assume that `PTA_CMD_GET_DEVICES_RPMB` always will be done before `PTA_CMD_GET_DEVICES_SUPP`, it's two independent actions in the kernel.

Reply:
I agree. And we also don't want to call `tee_rpmb_reinit()` each time. We need a dedicated function `tee_rpmb_needs_supplicant()` I think.